frontend/src/scenes/max/FeedbackPrompt.tsx

```diff
@@ -0,0 +1,237 @@
+import { useActions, useValues } from 'kea'
+import posthog from 'posthog-js'
+import { useEffect, useState } from 'react'
+
+import { IconDocument } from '@posthog/icons'
+import { LemonButton, LemonInput, LemonModal } from '@posthog/lemon-ui'
+
+import { SupportForm } from 'lib/components/Support/SupportForm'
+import { supportLogic } from 'lib/components/Support/supportLogic'
+
+import { MessageTemplate } from './messages/MessageTemplate'
+
+export type FeedbackRating = 'bad' | 'okay' | 'good' | 'dismissed' | 'implicit_dismiss'
+export type FeedbackTriggerType = 'message_interval' | 'random_sample' | 'manual' | 'retry' | 'cancel'
+
+function captureNegativeFeedback(
+    traceId: string | null,
+    triggerType: FeedbackTriggerType,
+    conversationId: string,
+    feedbackText?: string
+): void {
+    posthog.capture('posthog_ai_feedback_submitted', {
+        $ai_conversation_id: conversationId,
+        $ai_session_id: conversationId,
+        $ai_trace_id: traceId,
+        $ai_feedback_rating: 'bad',
+        $ai_feedback_trigger_type: triggerType,
+        $ai_feedback_text: feedbackText || null,
+    })
+}
+
+interface FeedbackPromptProps {
+    conversationId: string
+    traceId: string | null
+    triggerType: FeedbackTriggerType
+    onComplete: () => void
+    onRecordCooldown: () => void
+}
+
+/**
+ * Detailed feedback form shown after user clicks "Bad" rating.
+ * Allows text feedback submission or escalation to support ticket.
+ */
+export function FeedbackPrompt({
+    conversationId,
+    traceId,
+    triggerType,
+    onComplete,
+    onRecordCooldown,
+}: FeedbackPromptProps): JSX.Element {
+    const [feedbackText, setFeedbackText] = useState('')
```

Comment:
Most of these useStates and functions can be moved to kea, so the component is lean and you trigger all the functions there through actions